### PR TITLE
Adding beaker-task_helper and sort a-z

### PR DESCRIPTION
**Pull request by tphoney**

### CODE REVIEW

**Comment by tphoney** [comment]
@willmeek / @pmcmaw is there a neat way to test this ?

**Comment by pmcmaw** [comment]
I have carried out the following:
- checked these changes out locally
- built the gems locally `bundle exec exe/build-gems.rb`
- Updated puppet_conf gemfile to point at the local gems
- Removed beaker-task_helper from the Gemfile

I can confirm that the change worked and `beaker-task_helper` was included in the Gemfile.lock and distributed with these changes.

I was able to do a bundle install which confirms the formatting of the Gemfile is fine. I was also able to run agent only acceptance tests: `puppet_conf task
  puppet configuration file
    set/get a puppet configuration
    set/get a puppet configuration with section
Handing 'vrudrgq3osmya78.delivery.puppetlabs.net' back to vmpooler for VM destruction
Spent 0.72 seconds cleaning up

Finished in 1 minute 1.86 seconds (files took 1 minute 57.51 seconds to load)
2 examples, 0 failures`

**Comment by willmeek** [comment]
If it is of any value, I've written a unit test which tests that the default dev dependencies can be correctly parsed out, at least validates that everything is present and syntax correct

https://github.com/willmeek/puppet-module-gems/commit/f58c53f9ddc944849291e9d7ec839962a496be76#diff-daf5e8f75b6360cfef92a07d2416ad5c

**Comment by tphoney** [comment]
@bmjen I think this is good to merge. thanks to @pmcmaw and @willmeek

File: config/dependencies.yml [hunk 1]
```diff
@@ -6,6 +6,10 @@ dependencies:
       r2.4:
     dev:
       shared:
+        - gem: beaker-task_helper
```
Comment (bmjen) [inline config/dependencies.yml]: Anything beaker related needs to go to the `system` group, so we don't pull in beaker deps unnecessarily.

**Comment by tphoney** [comment]
@bmjen hopefully that is better

**Pull request merged**